xbmc/cores/VideoPlayer/AudioSinkAE.cpp

```diff
@@ -14,6 +14,8 @@
 #include "cores/AudioEngine/Interfaces/AE.h"
 #include "cores/AudioEngine/Utils/AEAudioFormat.h"
 #include "cores/AudioEngine/Utils/AEStreamData.h"
+#include "settings/Settings.h"
+#include "settings/SettingsComponent.h"
```

Comment:
This does not match the current design at all. You get a direct dependency to the settings into the AudioSink, where it does not belong. Audio Settings and on demand changes are changed message based and stored in the general audio engine configuration.

Long story short: One example is the "stream low volume noise" PR which uses the same infrastructure.